File: app-java/app/src/main/java/com/phunware/android/phunwareproducthomework/features/list/ZipCodeListViewModel.java
package com.phunware.android.phunwareproducthomework.features.list;

import com.phunware.android.phunwareproducthomework.WeatherApp;
import com.phunware.android.phunwareproducthomework.storage.ZipCodeStore;

import java.util.List;

import javax.inject.Inject;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

public class ZipCodeListViewModel extends ViewModel {
    @Inject
    ZipCodeStore zipCodeStore;

    public ZipCodeListViewModel() {
        WeatherApp.getApplicationComponent().inject(this);
    }

    private MutableLiveData<List<String>> zipCodes = null;

    public LiveData<List<String>> getZipCodes() {
        if (zipCodes == null) {
            zipCodes = new MutableLiveData<>();
        }

        loadZipCodes();

        return zipCodes;
    }

    private void loadZipCodes() {
        zipCodes.setValue(zipCodeStore.getZipCodes());
    }
}

File: app-kotlin/app/src/main/java/com/phunware/android/phunwareproducthomework/features/list/ZipCodeListViewModel.kt
package com.phunware.android.phunwareproducthomework.features.list

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import com.phunware.android.phunwareproducthomework.WeatherApp
import com.phunware.android.phunwareproducthomework.storage.ZipCodeStore
import javax.inject.Inject

class ZipCodeListViewModel : ViewModel() {

    @Inject
    lateinit var zipCodeStore: ZipCodeStore

    init {
        WeatherApp.getApplicationComponent().inject(this)
    }

    private var zipCodes: MutableLiveData<List<String>>? = null

    fun getZipCodes(): LiveData<List<String>> {
        if (zipCodes == null) {
            zipCodes = MutableLiveData()
        }

        loadZipCodes()

        return zipCodes!!
    }

    private fun loadZipCodes() {
        zipCodes!!.value = zipCodeStore.getZipCodes()
    }
}